### PR TITLE
{Compute} fix none check when list vmss extension.

### PR DESCRIPTION
**Pull request by yungezz**
Fix error in issues

#11533
#11785
Add None check to vmss.virtual_machine_profile
This checklist is used to make sure that common guidelines for a pull request are followed.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by jiasli** [comment]
@fengzhou-msft, we are seeing [this error](https://dev.azure.com/azure-sdk/public/_build/results?buildId=239301&view=logs&j=b21ded15-2ef6-5c2b-875f-d00fdf573f12&t=ae271903-525b-5a25-197b-c8c2e710d08d&l=908) more and more often

```
git checkout --progress --force bf1d02f8058230e6bfc720668f78d5f926d37de4
fatal: reference is not a tree: bf1d02f8058230e6bfc720668f78d5f926d37de4
##[error]Git checkout failed with exit code: 128
```

**Comment by jiasli** [comment]
I tried `az vm show` and `az vmss show` and saw `osProfile` and `virtualMachineProfile` appearing in the response json respectively. Under what circumstances will them be `null`?

**Comment by yungezz** [comment]
> I tried `az vm show` and `az vmss show` and saw `osProfile` and `virtualMachineProfile` appearing in the response json respectively. Under what circumstances will they be `null`?

ideally those properties are in results, while customers do see NoneType exception as showed in below issues:
#11533
#11785

**Comment by jiasli** [comment]
Thanks. I saw the exceptions in those two issues. While this PR solves them, I think it will still be great for us to find out why that happened in the first place. @qwordy

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 1]
```diff
@@ -1879,7 +1879,9 @@ def add_vm_secret(cmd, resource_group_name, vm_name, keyvault, certificate, cert
 
 def list_vm_secrets(cmd, resource_group_name, vm_name):
     vm = get_vm(cmd, resource_group_name, vm_name)
-    return vm.os_profile.secrets
+    if vm.os_profile:
+        return vm.os_profile.secrets
+    return None
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py]: `return None` will show nothing in the output.
how about `return []`

Comment (yungezz) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py]: Nice. actually I have same confusion on return value when result is None. sometimes az cli returns  [], sometimes None. Is there any general guideline?

**Comment by yungezz** [comment]
> Thanks. I saw the exceptions in those two issues. While this PR solves them, I think it will still be great for us to find out why that happened in the first place. If the `null` property is generated by CLI, maybe we can fix that in the create or update command. @qwordy

It could happen in parallel. Eventually it's not right to throw NoneType exception. While we can keep understanding users specific scenario, it takes time and to the end, specific to environments. because both commands are quite straightforward, CLI does nothing except wrapping rest call and output directly.

File: src/azure-cli/azure/cli/command_modules/vm/custom.py (hunk 1 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py]: In this command, the output is [] if there is no secret.
Let me search the doc.

**Comment by qwordy** [comment]
> 
> 
> Thanks. I saw the exceptions in those two issues. While this PR solves them, I think it will still be great for us to find out why that happened in the first place. If the `null` property is generated by CLI, maybe we can fix that in the create or update command. @qwordy

Adding null pointer checker is a general way to mitigate this kind of bug. I agree. We need to find out why it's null.

**Comment by fengzhou-msft** [comment]
I haven't identified the root cause for the git checkout error, seems random.  But I noticed we checkout source code for all tasks even those package/image testing tasks that don't need the source code. 

@haroldrandom do we have a way to skip checking out code in tasks?

File: src/azure-cli/azure/cli/command_modules/vm/custom.py (hunk 1 above)
Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py]: Tested with a VM and the results confirms @qwordy's comment:

```
az vm secret list -n {} -g {}
[]
```

**Comment by yonzhan** [comment]
@haroldrandom  please take a look at the question raised by @fengzhou-msft

**Comment by haroldrandom** [comment]
> I haven't identified the root cause for the git checkout error, seems random. But I noticed we checkout source code for all tasks even those package/image testing tasks that don't need the source code.
> 
> @haroldrandom do we have a way to skip checking out code in tasks?

As for randomly fail on `git checkout`, I haven't dig it out. It occurs often suddenly. Intuitively, ADO could make some changes.

As for skip-able git checkout, I searched Document of ADO, found some options like [Don't sync sources](https://docs.microsoft.com/zh-cn/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#dont-sync-sources) and [Shallow Fetch](https://docs.microsoft.com/zh-cn/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#shallow-fetch) might help. Nice suggestion to speed up cost time of CI, will dig it later.

**Comment by yungezz** [comment]
CI passed in Azure DevOps https://dev.azure.com/azure-sdk/public/_build/results?buildId=276518&view=results, seems status not sycing into github. Looking at it

**Pull request merged**